commit a4a85331f1d758c98b244f5be144421b40788bea
Author: Alberto97 <albertop2197@gmail.com>
Date:   Sun Dec 22 20:55:51 2013 +0100

    Revert Telephony: Add getLteOnGsmMode() method (2/2)

diff --git a/src/java/com/android/internal/telephony/BaseCommands.java b/src/java/com/android/internal/telephony/BaseCommands.java
index f224432..4155c97 100644
--- a/src/java/com/android/internal/telephony/BaseCommands.java
+++ b/src/java/com/android/internal/telephony/BaseCommands.java
@@ -747,12 +747,4 @@ public abstract class BaseCommands implements CommandsInterface {
     public int getRilVersion() {
         return mRilVersion;
     }
-
-    /**
-     * @hide
-     */
-    @Override
-    public int getLteOnGsmMode() {
-        return TelephonyManager.getLteOnGsmModeStatic();
-    }
 }
diff --git a/src/java/com/android/internal/telephony/CommandsInterface.java b/src/java/com/android/internal/telephony/CommandsInterface.java
index c89469b..f1ee23e 100644
--- a/src/java/com/android/internal/telephony/CommandsInterface.java
+++ b/src/java/com/android/internal/telephony/CommandsInterface.java
@@ -1623,12 +1623,6 @@ public interface CommandsInterface {
     public int getLteOnCdmaMode();
 
     /**
-     * Return if the current radio is LTE on GSM
-     * @hide
-     */
-    public int getLteOnGsmMode();
-
-    /**
      * Request the ISIM application on the UICC to perform the AKA
      * challenge/response algorithm for IMS authentication. The nonce string
      * and challenge response are Base64 encoded Strings.
diff --git a/src/java/com/android/internal/telephony/Phone.java b/src/java/com/android/internal/telephony/Phone.java
index c42beda..f4e26ea 100644
--- a/src/java/com/android/internal/telephony/Phone.java
+++ b/src/java/com/android/internal/telephony/Phone.java
@@ -1664,12 +1664,6 @@ public interface Phone {
     public int getLteOnCdmaMode();
 
     /**
-     * Return if the current radio is LTE on GSM
-     * @hide
-     */
-    public int getLteOnGsmMode();
-
-    /**
      * TODO: Adding a function for each property is not good.
      * A fucntion of type getPhoneProp(propType) where propType is an
      * enum of GSM+CDMA+LTE props would be a better approach.
diff --git a/src/java/com/android/internal/telephony/PhoneBase.java b/src/java/com/android/internal/telephony/PhoneBase.java
index ef65d11..3e5bc4a 100644
--- a/src/java/com/android/internal/telephony/PhoneBase.java
+++ b/src/java/com/android/internal/telephony/PhoneBase.java
@@ -1383,14 +1383,6 @@ public abstract class PhoneBase extends Handler implements Phone {
     }
 
     /**
-     * {@hide}
-     */
-    @Override
-    public int getLteOnGsmMode() {
-        return mCi.getLteOnGsmMode();
-    }
-
-    /**
      * Sets the SIM voice message waiting indicator records.
      * @param line GSM Subscriber Profile Number, one-based. Only '1' is supported
      * @param countWaiting The number of messages waiting, if known. Use
diff --git a/src/java/com/android/internal/telephony/PhoneFactory.java b/src/java/com/android/internal/telephony/PhoneFactory.java
index b3d5235..7e9dd4c 100644
--- a/src/java/com/android/internal/telephony/PhoneFactory.java
+++ b/src/java/com/android/internal/telephony/PhoneFactory.java
@@ -106,9 +106,6 @@ public class PhoneFactory {
                 if (TelephonyManager.getLteOnCdmaModeStatic() == PhoneConstants.LTE_ON_CDMA_TRUE) {
                     preferredNetworkMode = Phone.NT_MODE_GLOBAL;
                 }
-                if (TelephonyManager.getLteOnGsmModeStatic() != 0) {
-                    preferredNetworkMode = Phone.NT_MODE_LTE_GSM_WCDMA;
-                }
                 int networkMode = Settings.Global.getInt(context.getContentResolver(),
                         Settings.Global.PREFERRED_NETWORK_MODE, preferredNetworkMode);
                 Rlog.i(LOG_TAG, "Network Mode set to " + Integer.toString(networkMode));
diff --git a/src/java/com/android/internal/telephony/PhoneProxy.java b/src/java/com/android/internal/telephony/PhoneProxy.java
index ad4090f..ff034f8 100644
--- a/src/java/com/android/internal/telephony/PhoneProxy.java
+++ b/src/java/com/android/internal/telephony/PhoneProxy.java
@@ -1168,14 +1168,6 @@ public class PhoneProxy extends Handler implements Phone {
         return mActivePhone.getLteOnCdmaMode();
     }
 
-    /**
-     * {@hide}
-     */
-    @Override
-    public int getLteOnGsmMode() {
-        return mActivePhone.getLteOnGsmMode();
-    }
-
     @Override
     public void setVoiceMessageWaiting(int line, int countWaiting) {
         mActivePhone.setVoiceMessageWaiting(line, countWaiting);
