commit 9270586f3f7b187605a63aabb84e38da8f5ae8ac
Author: KINGbabasula <kingbabasula@gmail.com>
Date:   Mon Dec 23 09:36:46 2013 +0100

    Revert Telephony: Add getLteOnGsmMode() method
    
    Change-Id: I2127b3f451e4188e811922b8eb8102465c3f011c

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index b4bc86e..26cd8e4 100644
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -901,8 +901,4 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
     public int getLteOnCdmaMode() {
         return mPhone.getLteOnCdmaMode();
     }
-
-    public int getLteOnGsmMode() {
-        return mPhone.getLteOnGsmMode();
-    }
 }
